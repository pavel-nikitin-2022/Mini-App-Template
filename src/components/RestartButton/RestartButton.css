@keyframes bubling {
  from {
    transform: translateY(0px);
    opacity: 0;
  }

  to {
    transform: translateY(-50px);
    opacity: 1;
  }
}

.RestartButton {
  position: absolute !important;
  bottom: 0px;
  width: 90% !important;
  max-width: 374px !important;
  margin: 0 auto !important;
  animation: bubling 0.75s ease-in-out forwards;
}
