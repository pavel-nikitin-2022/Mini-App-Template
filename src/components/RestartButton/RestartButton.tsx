import React from 'react'
import { Button } from '@vkontakte/vkui'
import './RestartButton.css'

type ResultButtonsProps = {
  onRestart: () => void
}

let RestartButton: React.FC<ResultButtonsProps> = ({ onRestart }) => {
  return (
    <Button className="RestartButton" stretched={true} onClick={onRestart}>
      Вернуться
    </Button>
  )
}

RestartButton = React.memo(RestartButton)

export { RestartButton }
