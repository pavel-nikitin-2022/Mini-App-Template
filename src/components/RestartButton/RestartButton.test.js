import React from 'react'
import { RestartButton } from './RestartButton'
import { shallow } from 'enzyme'

const setUp = (props) => shallow(<RestartButton {...props} />)

describe('RestartButton', () => {
  it('Компонент должен содержать класс RestartButton', () => {
    const component = setUp({ onRestart: () => null })
    const wrapper = component.find('.RestartButton')
    expect(wrapper.length).toBe(1)
  })

  it('Соответствие снимку', () => {
    const component = setUp({ onRestart: () => null })
    expect(component).toMatchSnapshot()
  })
})
