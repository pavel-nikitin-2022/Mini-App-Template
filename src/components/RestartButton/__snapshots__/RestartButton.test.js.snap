// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestartButton Соответствие снимку 1`] = `
<div
  className="RestartButton"
>
  <Button
    onClick={[Function]}
    stretched={true}
  >
    Вернуться
  </Button>
</div>
`;
